modules/tinymce/src/core/main/ts/dom/Empty.ts

```diff
@@ -52,6 +52,10 @@ const isEmptyNode = (targetNode: Node, skipBogus: boolean): boolean => {
           continue;
         }
       }
+      if (NodeType.isContentEditableTrue(node)
+        && Traverse.parentElement(SugarElement.fromDom(node)).exists((parent) => !ContentEditable.isEditable(parent))) {
+        return false;
+      }
```

Comment:
I don't think you need to go though sugar and options in this case. I think this should do the same thing.
```suggestion
      if (NodeType.isContentEditableTrue(node) && !ContentEditable.isEditable(node.parentNode)) {
        return false;
      }
```

Reply:
`ContentEditable.isEditable` requires the first argument to be SugarElement.

Reply:
Right I read that as `dom.isEditable` some how. I think at least the `if` statement should be split up into having a that long statement in there looks a bit odd. Maybe make a isEditableParent predicate of the inline thing just to split it up.

Reply:
added `hasNonEditableParent` predicate